### PR TITLE
GoWatchIt: Code cleanup and refractor

### LINKED ISSUE #2213: GoWatchIt: Code cleanup and refactor
- [ ] should not call isRelevant directly (we should be using a relevancy block if possible)
- [ ] function getUrl is a one-liner, and only used once - not necessary
- [ ] random variables declared all over the place moved to the top
- [ ] fix everything that says "This probably shouldn't be the way to do it, but it works for now."

Note: Some of this may require internal changes from the DDG Staff. Please let us know if you need anything and we'll see what we can do

### PR DESCRIPTION
**Pull request by sahildua2305**
###### What does your Pull Request do (check all that apply)?

- [x] Improvement
    - [x] Bug fix: **`GoWatchIt: Fix #2213`**
    - [ ] Enhancement: **`{IA Name}: {Description of Improvements}`**

###### Description of changes

- [x] functions should be declared as variables and all in one place, not all over the file
- [x] function getUrl is a one-liner, and only used once - not necessary
- [x] random variables declared all over the place moved to the top
- [x] numbers as keys in a JSON structure should be quoted

###### Which issues (if any) does this fix?

Fixes #2213 

###### People to notify (@mention interested parties)
@moollaza 

---

Instant Answer Page: https://duck.co/ia/view/go_watch_it

[Maintainer](http://docs.duckduckhack.com/maintaining/guidelines.html): @plexusent

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by sahildua2305** [comment]
@moollaza I just noticed that this IA is being maintained by an organization. So can't assign any person for feedback.

Please suggest more changes. I tried my best to make all changes necessary to make the code properly formatted.

File: share/spice/go_watch_it/go_watch_it.js [hunk 1]
```diff
@@ -1,6 +1,132 @@
...
+        "10": 1,
+        // Filmmaker Direct Streaming
+        "46": 1,
+        // DirecTV
+        "113": 1
+    };
+
+    var provider_icons = {
+        "1": {
+            dark: DDG.get_asset_path('go_watch_it', path + 'netflix'),
```
Comment (moollaza) [inline share/spice/go_watch_it/go_watch_it.js]: It doesn't seem like there's any need for us to call `DDG.get_asset_path` in each of these? What about an array, and we call `DDG.get_asset_parth()` below, when the path is actually needed?

Something like...

```js
var providers = {
 1: [ 'netflix', 'netflix'],
 2: [ 'netflix', 'netflix'],
 3: [ 'itunes', 'itunes-alt'],
 ...
};
```

File: share/spice/go_watch_it/go_watch_it.js [hunk 2]
```diff
@@ -56,17 +186,13 @@
...
-        var foundNetflix = false;
-
         Spice.add({
             id: "go_watch_it",
             name: "Watch",
             data: watchable.availabilities,
             meta: {
                 sourceName: 'GoWatchIt',
-                sourceUrl: getUrl(watchable),
+                sourceUrl: 'http://gowatchit.com' + (watchable.url ? watchable.url : ""),
```
Comment (moollaza) [inline share/spice/go_watch_it/go_watch_it.js]: let's use HTTPS

File: share/spice/go_watch_it/go_watch_it.js [hunk 3]
```diff
@@ -304,26 +300,23 @@
...
                 format_line = "Available";
             }
         }
-        
         return format_line;
     });
-    
+
     Spice.registerHelper("gwi_buyOrRent", function (buy_line, rent_line, options) {
         if (buy_line && buy_line !== "" && !/\$0\.00/.test(buy_line) && /\$/.test(buy_line)) {
```
Comment (moollaza) [inline share/spice/go_watch_it/go_watch_it.js]: - Can we simplify this to 1 regex with an optional capture?
- We should cache the compiled regex in a variable

File: share/spice/go_watch_it/go_watch_it.js (hunk 1 above)
Comment (sahildua2305) [inline share/spice/go_watch_it/go_watch_it.js]: Very good point! @moollaza Thanks :)

**Comment by sahildua2305** [comment]
@moollaza Please review the changes.

File: share/spice/go_watch_it/go_watch_it.js [hunk 4]
```diff
@@ -1,6 +1,69 @@
...
+        // Movies on Demand
+        "10": 1,
+        // Filmmaker Direct Streaming
+        "46": 1,
+        // DirecTV
+        "113": 1
+    };
+
+    var provider_icons = {
+        "1": ['netflix', 'netflix'],
```
Comment (moollaza) [inline share/spice/go_watch_it/go_watch_it.js]: nit: we might as well sort this list numerically?

Comment (sahildua2305) [inline share/spice/go_watch_it/go_watch_it.js]: Pushed a quick fix, please check!

**Comment by moollaza** [comment]
@sahildua2305 Thanks! This LGTM :+1:

**Pull request merged**